server/lib/tuist_web/live/account_settings_live.ex

```diff
@@ -126,6 +131,52 @@ defmodule TuistWeb.AccountSettingsLive do
     {:noreply, socket}
   end
 
+  def handle_event(
+        "create_cache_endpoint",
+        %{"account_cache_endpoint" => %{"url" => url}},
+        %{assigns: %{selected_account: selected_account}} = socket
+      ) do
+    case Accounts.create_account_cache_endpoint(selected_account, %{url: url}) do
+      {:ok, _endpoint} ->
+        cache_endpoints = Accounts.list_account_cache_endpoints(selected_account)
+        add_cache_endpoint_form = to_form(AccountCacheEndpoint.create_changeset(%{}))
+
+        socket =
+          socket
+          |> assign(cache_endpoints: cache_endpoints)
+          |> assign(add_cache_endpoint_form: add_cache_endpoint_form)
+          |> push_event("close-modal", %{id: "add-cache-endpoint-modal"})
+
+        {:noreply, socket}
+
+      {:error, %Ecto.Changeset{} = changeset} ->
+        {:noreply, assign(socket, add_cache_endpoint_form: to_form(changeset))}
+    end
+  end
+
+  def handle_event(
+        "delete_cache_endpoint",
+        %{"id" => endpoint_id},
+        %{assigns: %{selected_account: selected_account}} = socket
+      ) do
+    endpoint = Accounts.get_account_cache_endpoint!(endpoint_id)
+    {:ok, _} = Accounts.delete_account_cache_endpoint(endpoint)
+    cache_endpoints = Accounts.list_account_cache_endpoints(selected_account)
```

Comment:
What about making `list_account_cache_endpoints` return `{:ok, results}` to pattern match against that and let errors blow up?